### PR TITLE
simplify conditions

### PR DESCRIPTION
**Pull request by aristath**
This is an addition to https://github.com/WordPress/gutenberg/pull/35034, simplifying and combining conditions to make things easier to read & understand.

cc @carolinan

### CODE REVIEW

**Comment by carolinan** [comment]
The PR works well for the different scenarios listed in the previous PR.

Are the test failing only because the PR is not against trunk, or?

**Comment by aristath** [comment]
> Are the test failing only because the PR is not against trunk, or?

Yeah, it's because it's not against `trunk` 🤷 
It's safe to ignore failing tests here

File: packages/block-library/src/search/index.php [hunk 1]
```diff
@@ -43,37 +43,17 @@ function render_block_core_search( $attributes ) {
...
-		} else {
-			// If the label text has been removed and toggled off, add a visually hidden label with default text.
-			$label_markup = sprintf(
-				'<label for="%s" class="wp-block-search__label screen-reader-text">%s</label>',
-				$input_id,
-				__( 'Search' )
-			);
-		}
+	$label_markup = sprintf(
+		'<label for="%s" class="wp-block-search__label screen-reader-text">%s</label>',
```
Comment (Mamaduka) [inline packages/block-library/src/search/index.php]:
```suggestion
		'<label for="%1$s" class="wp-block-search__label screen-reader-text">%2$s</label>',
```

I think there's WPCS for missing argument numbers.

File: packages/block-library/src/search/index.php [hunk 2]
```diff
@@ -43,37 +43,17 @@ function render_block_core_search( $attributes ) {
...
-			);
-		}
+	$label_markup = sprintf(
+		'<label for="%1$s" class="wp-block-search__label screen-reader-text">%2$s</label>',
+		$input_id,
+		empty( $attributes['label'] ) ? __( 'Search' ) : $attributes['label']
+	);
+	if ( $show_label && ! empty( $attributes['label'] ) ) {
+		$label_markup = sprintf(
+			'<label for="%s" class="wp-block-search__label">%s</label>',
```
Comment (Mamaduka) [inline packages/block-library/src/search/index.php]:
```suggestion
			'<label for="%1$s" class="wp-block-search__label">%2$s</label>',
```

Sorry missed this one.

File: packages/block-library/src/search/index.php [hunk 3]
```diff
@@ -43,37 +43,17 @@ function render_block_core_search( $attributes ) {
...
-				$input_id,
-				__( 'Search' )
-			);
-		}
+	$label_markup = sprintf(
+		'<label for="%1$s" class="wp-block-search__label screen-reader-text">%2$s</label>',
+		$input_id,
+		empty( $attributes['label'] ) ? __( 'Search' ) : $attributes['label']
+	);
+	if ( $show_label && ! empty( $attributes['label'] ) ) {
```
Comment (Mamaduka) [inline packages/block-library/src/search/index.php:51]: If `$show_label` is true, but `$attributes['label']` is empty this logic won't display default label "Search". Unless I'm reading conditional wrong.

**Comment by Mamaduka** [comment]
I love the simplified logic here.

~~I think we can also apply `screen-reader-text` based on `$show_label` and just have a single `sprintf` for markup. What do you think?~~

Ignore my last suggestion. @carolinan clarified few things for me.

File: packages/block-library/src/search/index.php (hunk 3 above)
Comment (carolinan) [reply packages/block-library/src/search/index.php:51]: I don't think it should show it, because the user did delete the visible text.

**Review by Mamaduka — APPROVED** [review]

**Pull request merged**